packages/ethereum/redux/thunks.js

```diff
@@ -4,27 +4,23 @@
  */
 
 import * as Keychain from 'react-native-keychain'
-import { getAddress } from '@chronobank/bitcoin/utils'
-import { login, savePrivateKey } from '@chronobank/session/redux/actions'
-import { bitcoinCreateWallet } from '@chronobank/bitcoin/redux/actions'
-import { encryptWallet, decryptWallet } from '../utils'
+import { savePrivateKey } from '@chronobank/session/redux/actions'
+import { encryptWallet, createEthWallet } from '../utils'
 import { ethereumCreateWallet } from './actions'
 
 // eslint-disable-next-line import/prefer-default-export
 export const createAccount = (mnemonic, password) => async (dispatch) => {
   try {
-    const encryptedWallet = await encryptWallet(mnemonic, password)
-    const decryptedWallet = await decryptWallet(encryptedWallet, password)
+    const decryptedWallet = createEthWallet(mnemonic)
     const ethAddress = decryptedWallet.address
-    const bitcoinAddress = getAddress(decryptedWallet.privateKey)
-
+    
     await Keychain.setInternetCredentials(ethAddress, ethAddress, password)
-
+    
+    const encryptedWallet = await encryptWallet(decryptedWallet, password)
     dispatch(ethereumCreateWallet(ethAddress, encryptedWallet))
-    dispatch(savePrivateKey(decryptedWallet.privateKey))
 
-    dispatch(bitcoinCreateWallet(ethAddress, bitcoinAddress))
-    dispatch(login(ethAddress))
+    dispatch(savePrivateKey(decryptedWallet.privateKey))
```

Comment:
Thunk createAccount must create account. We should not saveProvateKey inside this thunk.
We need to call `dispatch(savePrivateKey(decryptedWallet.privateKey))` right after call createAccount thunk in appropriate place (we will discuss it)